tools/tpm2_certifycreation.c

```diff
@@ -227,31 +227,15 @@ static tool_rc process_certifycreation_input(ESYS_CONTEXT *ectx,
     /*
      * Qualifier data is optional. If not specified default to 0
      */
-    if (!ctx.policy_qualifier_path) {
-        goto process_certifycreation_input_out;
+    if (!ctx.policy_qualifier_data) {
+        return tool_rc_success;
     }
 
-    unsigned long file_size = 0;
-    result = files_get_file_size_path(ctx.policy_qualifier_path, &file_size);
-    if (!result || file_size == 0) {
-        return tool_rc_general_error;
-    }
-
-    *policy_qualifier = malloc(file_size + sizeof(uint16_t));
```

Comment:
@idesai this code is wild.
1. if you need to malloc a TPM2B_* data type, just malloc based on the sizeof of the type. We're not counting the few bytes it wastes.
2. Why did you even malloc it? There was no point in complicating the flow with needing to free and dealing with oom. Just stack allocate and pass a pointer, which is funny, as a lot of the other parameter structures worked like that.